src/internal/cluster/zarf.go

```diff
@@ -31,13 +31,13 @@ func (c *Cluster) GetDeployedZarfPackages() ([]types.DeployedPackage, error) {
 	for _, secret := range secrets.Items {
 		var deployedPackage types.DeployedPackage
 		err := json.Unmarshal(secret.Data["data"], &deployedPackage)
+		// Don't break out of the loop if we can't unmarshal a secret, just log it
 		if err != nil {
-			message.Warnf("Unable to unmarshal package secret")
-
-			return deployedPackages, err
```

Comment:
I think we should save the error and return it later. If we are unable to unmarshal a secret then something is probably really wrong with either Zarf or the secret. Simply warning it probably isn't enough.


I think it makes sense to continue parsing through the secrets so that the caller can get a list of all package secrets we were able to parse and error(s?) for the bad package secrets and decide how to handle it themselves.

Reply:
@mike-winberry Is this error something you ran into yourself?

Reply:
Yeah so I think it should be tagged for https://github.com/defenseunicorns/zarf/issues/1312. Cause right now basically it just bailed and nothing was happening on the ui, so not even the working packages were showing up as a return for this method. And since there is no error handling for the front end right now, I still need it to show the good packages and then log the failures until there is something in place on the front end to handle those errors. But I can go ahead and at least capture the errors for the front end now even if the logic isn't in place to handle them.

Reply:
If theres an error when deploying the init package it kinda borks the whole ui unless I ignore the un-marshalling error.

Reply:
Its something I need to dig deeper into, but should be handled in #1312. I agree that we shouldn't be ignoring, Ill rewrite to add least pass the errors found back from this routine so they will be there for capturing in #1312.

Reply:
It mostly came up because I was having to force errors for this story in order to ensure they were being captured in the deploy page, then I started running into the unmarshalling error that was causing any package after a broken package to not be returned from this routine.

Reply:
![Screenshot 2023-06-21 at 9 52 57 PM](https://github.com/defenseunicorns/zarf/assets/86802655/7b11a4e8-a077-4512-b361-8bd5ecb7e101)

Reply:
The specifics are logged to the log file for unmarshaling errors.